google/cloud/spanner_v1/database.py

```diff
@@ -1040,6 +1052,39 @@ def __exit__(self, exc_type, exc_val, exc_tb):
             self._database._pool.put(self._session)
 
 
+class MutationGroupsCheckout(object):
+    """Context manager for using mutation groups from a database.
+
+    Inside the context manager, checks out a session from the database,
+    creates mutation groups from it, making the groups available.
+
+    Caller must *not* use the object to perform API requests outside the scope
+    of the context manager.
+
+    :type database: :class:`~google.cloud.spanner_v1.database.Database`
+    :param database: database to use
+    """
+
+    def __init__(self, database):
+        self._database = database
+        self._session = self._mutation_groups = None
+
+    def __enter__(self):
+        """Begin ``with`` block."""
+        session = self._session = self._database._pool.get()
+        return MutationGroups(session)
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """End ``with`` block."""
+        if isinstance(exc_val, NotFound):
+            # If NotFound exception occurs inside the with block
+            # then we validate if the session still exists.
```

Comment:
Is there any reason behind creating a new session here? since this is already handled by the pool when a session is checked out from the pool.
https://github.com/googleapis/python-spanner/blob/e5acb568c276063d45a6db58d0744e8e59a6afce/google/cloud/spanner_v1/pool.py#L235-L236

Reply:
Yes, please see https://github.com/googleapis/python-spanner/pull/718.